src/ResourceManager/Version2016_09_01/ArgumentCompleters/ArgumentCompleterHelper.cs

```diff
@@ -0,0 +1,106 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+namespace Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+    using Microsoft.Rest.Azure;
+
+    /// <summary>
+    /// Provides helper functions and classes that other argument completers can use.
+    /// </summary>
+    public static class ArgumentCompleterHelper
+    {
+        /// <summary>
+        /// Builds the actual ps script for argument completer.
+        /// </summary>
+        public class ScriptBuilder
```

Comment:
Why is this non-static class encapsulated in a static class?  Why not ximply have a ScriptBuilder class.  Normally classes are only encapsulated to implement the AbstractFactory pattern, singleton pattern, , or similar patterns where the detail of the underlying structure is necessary